File: frontend/src/microservices/advertising/objects/GenerateAdvertisementImageCandidatesRequest.ts
// 本文件定义 advertising 模块的 `GenerateAdvertisementImageCandidatesRequest`，作为请求参数并提供 JSON 编解码。

export type GenerateAdvertisementImageCandidatesRequest = {
  prompt: string
  supportingCopy?: string | null
  tone?: string | null
  resourceLabel?: string | null
  advertisementKind?: string | null
  imageFactoryKind?: string | null
  transparentBackground?: boolean | null
  width?: number | null
  height?: number | null
  candidateCount?: number | null
  avoidText?: string | null
}

export const generateAdvertisementImageCandidatesRequestFromJson = (json: string): GenerateAdvertisementImageCandidatesRequest =>
  JSON.parse(json) as GenerateAdvertisementImageCandidatesRequest

export const generateAdvertisementImageCandidatesRequestToJson = (value: GenerateAdvertisementImageCandidatesRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/advertising/objects/GenerateAdvertisementImageCandidatesRequest.scala
// 本文件定义 advertising 模块的 GenerateAdvertisementImageCandidatesRequest，作为后端与前端同名的数据模型，并提供 JSON codec。
package com.typesafe.travel.advertising.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class GenerateAdvertisementImageCandidatesRequest(
    prompt: String,
    supportingCopy: Option[String],
    tone: Option[String],
    resourceLabel: Option[String],
    advertisementKind: Option[String],
    imageFactoryKind: Option[String],
    transparentBackground: Option[Boolean],
    width: Option[Int],
    height: Option[Int],
    candidateCount: Option[Int],
    avoidText: Option[String]
)

object GenerateAdvertisementImageCandidatesRequest:
  given sourceEncoder: Encoder[GenerateAdvertisementImageCandidatesRequest] = deriveEncoder
  given sourceDecoder: Decoder[GenerateAdvertisementImageCandidatesRequest] = deriveDecoder

